### ISSUE TITLE
A non-access logfile is set then it lead to the plugin running failed.

### ISSUE THREAD

**Issue by Zim-Inn** [haskell/haskell-language-server#3561]
## Is your enhancement request related to a problem? Please describe.
![image](https://user-images.githubusercontent.com/60260187/232238717-6a7afa7d-bfa8-4203-8fe4-2afa12fe1ef4.png)

For example, I set the logfile to "D:\boo\hlslogDir", it is a directory. And the I met the problem:
![AUK@HRQ PV7S8}55LS6C(8T](https://user-images.githubusercontent.com/60260187/232238812-d010461f-5387-436e-ac1d-527d6bbb36ed.png)

When I set the logFile setting to null, it worked.
![I7FX3GX@GS~D61LXL (}5U4](https://user-images.githubusercontent.com/60260187/232238898-0fd7448b-112a-4dcc-a744-0a3acb4bddb7.png)

## Describe the solution you'd like
So I think it should has a warning hint on the setting bar, or it should be mentioned at Readme's F&Q

**Comment by July541** [comment]
@fendor Looks like an enhancement in vscode-haskell?

**Comment by fendor** [comment]
Yep, updating the documentation to reflect the fact that the directory must exist (did I get that right?) seems very sensible to me.

Additionally, maybe we don't want to crash as HLS in that situation?

**Comment by July541** [comment]
> Yep, updating the documentation to reflect the fact that the directory must exist (did I get that right?) seems very sensible to me.

Or we can show a warning dialog?

> Additionally, maybe we don't want to crash as HLS in that situation?

Make sense to me. Maybe we can check the path before passing to hls? Not sure if worth to take.

**Comment by Zim-Inn** [comment]
I test the problem again. The plugin crashed just because I set a log directory path but it need a "log file path" indeed.
So I think you can create logfile automaticlly when the dir was set. Or warning that logFile setting must be set to a exist file path.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Lei ***@***.***&gt;; 
发送时间: 2023年4月16日(星期天) 下午4:00
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [haskell/haskell-language-server] A non-access logfile is set then it lead to the plugin running failed. (Issue #3561)





 
@fendor Looks like an enhancement in vscode-haskell?
